From c1986b178ec7ead9937d52a72c341887fa10837d Mon Sep 17 00:00:00 2001
From: sergeysdk1 <sdk@fromru.com>
Date: Mon, 27 Mar 2017 22:37:36 +0300
Subject: [PATCH] Fix parameters in camera

For hight frame rate mode add value "off", if not exist

Change-Id: Ie5cdb53d8c9d21d3f8f183b8062a03851a704a24
---
 src/com/android/camera/CameraSettings.java | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 85faf07..bac3985 100644
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -1342,6 +1342,17 @@ public class CameraSettings {
             return null;
         }
 
+        boolean hasOff = false;
+        for (String highFrameRateMode : supportedModes) {
+            if (highFrameRateMode.equals("off")) {
+                hasOff = true;
+                break;
+            }
+        }
+
+        if (!hasOff)
+            supported.add("off");
+
         for (String highFrameRateMode : supportedModes) {
             if (highFrameRateMode.equals("off")) {
                 supported.add(highFrameRateMode);
-- 
2.7.4

